### PR TITLE
feat(ui-components): split repository

### PR DESCRIPTION
**Pull request by shortcuts**
Follow up of https://github.com/algolia/recommend/pull/48

**Summary**

This PR groups everything related to the `ui-component-horizontal-sider` packages which will now be available here: https://github.com/algolia/ui-components.

**Result**

This PR won't be merged, but I'll create the new repository (https://github.com/algolia/ui-components) from it.

### CODE REVIEW

File: examples/demo/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  imageLink: string;
+  name: string;
+  price: number;
+  _score: number;
+  reviewScore: number;
+  reviewCount: number;
+  url: string;
+};
+
+const items: ProductHit[] = [
```
Comment (francoischalifour) [inline examples/demo/src/App.tsx]: We can move these items to a separate file to see and edit the `App` file more easily.

File: examples/demo/src/App.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    name: '5620 G-Star Elwood Staq 3D Mid Waist Skinny Jeans',
+    objectID: 'D06194-8968-8960',
+    price: 190,
+    reviewCount: 53,
+    reviewScore: 6,
+    url: 'women/jeans/d06194-8968-8960',
+  },
+];
+
+function Hit({ item }: { item: ProductHit }) {
```
Comment (francoischalifour) [inline examples/demo/src/App.tsx]: We can declare a `HitProps` type.

**Review by francoischalifour — APPROVED** [review]
Awesome, thank you! Feel free to create the `algolia/ui-components` repo based on that branch.

File: examples/demo/src/App.tsx (hunk 1 above)
Comment (shortcuts) [reply examples/demo/src/App.tsx]: Done in https://github.com/algolia/recommend/pull/50/commits/f5438dd5a69aeeff9227ac8000a66c1cdd96b61f

File: examples/demo/src/App.tsx (hunk 2 above)
Comment (shortcuts) [reply examples/demo/src/App.tsx]: Done in https://github.com/algolia/recommend/pull/50/commits/f5438dd5a69aeeff9227ac8000a66c1cdd96b61f

**Comment by shortcuts** [comment]
The new repository is live: https://github.com/algolia/ui-components

**Comment by shortcuts** [comment]
Closing this PR as it was just a reference to create https://github.com/algolia/ui-components

**Pull request closed without merging**